### DISCUSSION TITLE
ARDUCAM PICO4ML and NinaWiFi

### DISCUSSION THREAD

**Discussion by fethicandan** [earlephilhower/arduino-pico#2091] [General]
Hi,

I am trying to communicate with pico4ml and NinaWifi Module. Actually I am using Arduino Nano Connect hardware and Pico4ml system. Beucase I wanted to use IMU and Camera simultaneously. Camera and IMU are working on RP2040 Connect using pico4ml, but I want to use WiFiNnina module as well. 

Do you have any suggestions ?

P.S. I tried to add HM01B0 camera library into Arduino connect library but it is not working. Moreover, I used WifiNina Generic library, the system didn't recognise the SPI WifiNina module.

**Comment by JAndrassy** [comment]
you could try the esp-hosted firmware for NINA with the WiFi library in this core
https://github.com/earlephilhower/arduino-pico/tree/master/libraries/lwIP_ESPHost (follow the link to ESPHost library. it has the fw built for Nano Connect)

**Comment by fethicandan** [comment]
> you could try the esp-hosted firmware for NINA with the WiFi library in this core https://github.com/earlephilhower/arduino-pico/tree/master/libraries/lwIP_ESPHost (follow the link to ESPHost library. it has the fw built for Nano Connect)

Which board may I choose? Raspberry Pi Pico or RP2040? 
Even if I choose one of them. How can I use the HM01B0 camera and ICM20498 IMU? Beucase of these libraries I selected pico4ml. Otherwise, I can use directly Arduino RP2040 Connect with NinaWifi, am I right?

**Comment by JAndrassy** [comment]
sorry, it looks that I don't understand your problem

**Comment by fethicandan** [comment]
No worries.

My problem looks simple but I don't know how to solve that. I have Pico4ML and Arduino RP2040 Connect Boards. I bought an HM01B0 camera module. 

First, I tried the Pico4ML camera example with Pico4ML board. It works.
Then, I tried the Pico4ML camera example with Arduino connect board (defined as Pico4ML). It works.

But, I couldn't find any library to define NinaWifi module.

**Reply by earlephilhower** [reply]
There is no WifiNINA class possible outside of the MBED core.  You can use the wifi chip by doing what @JAndrassy said, in which case you'd be using the ESPHost library and would need to flash the onboard ESP appropriately.

**Comment by fethicandan** [comment]
> There is no WifiNINA class possible outside of the MBED core. You can use the wifi chip by doing what @JAndrassy said, in which case you'd be using the ESPHost library and would need to flash the onboard ESP appropriately.

Thank you so much reply my email. I checked something about NinaWifi. I found one of the libraries: ''Nina Generic Library'. It works with Raspberry Pi Pico and more hardware.

I will try to change something.

Actually I am doing like that:
Hardware: Arduino nano RP2040 connect
Hardware Definition: Arducam Pico4ML

Because I want to connect it with camera and nina WiFi on RP2040. Do you have any suggestions for that?
 
P.S. I will try to design a new PCB. Therefore I have some limitations: Size of board and power consumption.

**Comment by JAndrassy** [comment]
WiFiNINA works with any Arduino. just create boards.local.txt with -D for pins. for Nano Connect the pins are
```
<board>.build.extra_flags=-DPIWIFI=(SPI1) -DSPIWIFI_SS=(9u) -DSPIWIFI_ACK=(10u) -DSPIWIFI_RESET=(3u) -DNINA_GPIO0=(2u)
```

**Reply by earlephilhower** [reply]
These defines can be set in the boards.txt file for the Arduino side (and I think the p.io)

**Reply by JAndrassy** [reply]
Nano Connect has them in pins_arduino.h. OP wants them for Arducam Pico4ML because they want to use Nano Connect as Arducam Pico4ML for some reason (X->Y problem?)

**Reply by earlephilhower** [reply]
Ah, great. Thanks. I'm traveling so looking through code on my phone is hard.  Guess there's nothing to do here, will close the issue.

**Comment by fethicandan** [comment]
> WiFiNINA works with any Arduino. just create boards.local.txt with -D for pins. for Nano Connect the pins are
> 
> ```
> <board>.build.extra_flags=-DPIWIFI=(SPI1) -DSPIWIFI_SS=(9u) -DSPIWIFI_ACK=(10u) -DSPIWIFI_RESET=(3u) -DNINA_GPIO0=(2u)
> ```


Thank you so much. Could you explain a little bit more? Because I don't have any experience at this part.

**Reply by JAndrassy** [reply]
https://arduino.github.io/arduino-cli/0.35/platform-specification/#boardslocaltxt
